swarm.go

```diff
@@ -115,25 +115,21 @@ func NewSwarm(config *api.Config, mockStore *mock.NodeStore) (self *Swarm, err e
 	if config.SwapEnabled {
 		// for now, Swap can only be enabled in a whitelisted network
 		if self.config.NetworkID != swap.AllowedNetworkID {
-			return nil, fmt.Errorf("swap can only be enabled under Network ID %d, found Network ID %d instead", swap.AllowedNetworkID, self.config.NetworkID)
+			return nil, fmt.Errorf("swap can only be enabled under BZZ Network ID %d, found Network ID %d instead", swap.AllowedNetworkID, self.config.NetworkID)
 		}
-		// if Swap is enabled, we MUST have a contract API
-		if self.config.SwapBackendURL == "" {
-			return nil, errors.New("swap enabled but no contract address given; fatal error condition, aborting")
-		}
-		log.Info("connecting to SWAP API", "url", self.config.SwapBackendURL)
-		self.backend, err = ethclient.Dial(self.config.SwapBackendURL)
-		if err != nil {
-			return nil, fmt.Errorf("error connecting to SWAP API %s: %s", self.config.SwapBackendURL, err)
-		}
-
-		// initialize the balances store
-		swapStore, err := state.NewDBStore(filepath.Join(config.Path, "swap.db"))
```

Comment:
@holisticode , I think this was an error:
We use `config.Path` here, but `config.Path` is not initialized. Please see `swarm:124` and `swap/swap:114` about how it is  initialized now (using the `path` as provided by the configuration).

Am I correct that this was not intended and that we should use `self.config.Path`?

Reply:
Well `config` comes from the function parameter which we assign to `self.config` on L:110, so it is basically the same, or what am I missing?

Having said that I do believe that it is nicer to use `self.config.Path` because it is more consistent.

Reply:
You are right, it is the same indeed.

Reply:
Thank, I did oversee that config was a function parameter parameter